apps/aecore/src/aec_accounts.erl

```diff
@@ -129,6 +132,13 @@ type(#account{ ga_contract = C })         -> contract = aeser_id:specialize_type
 is_payable(#account{ flags = 0 }) -> true;
 is_payable(#account{ flags = N }) -> not get_flag(non_payable, N).
 
+-spec set_payable(account(), boolean()) -> account().
+set_payable(Acc = #account{ flags = N }, false) ->
+    Acc#account{ flags = (N bsr 1) bsl 1 };
+set_payable(Acc, true) ->
+    #account{ flags = N } = set_payable(Acc, false),
+    Acc#account{ flags = N + ?FLAG_NON_PAYABLE_VALUE }.
```

Comment:
The first clause should use `?FLAG_NON_PAYABLE_VALUE` (or rather `?FLAG_NON_PAYABLE` which isn't there).

We need to fix flag handling once and for all, and we also need to handle `non_payable` when creating a contract where the account pre-exists. Let's do that in another issue (#3579)

Reply:
Now should be nicer